File: src/react/src/models/step.ts
import { Job } from './job';
import { Server } from './server';

export type Step = {
    stepId: string;
    jobId: string;
    type: string;
    name: string;
    description: string;
    order: number;
    settings: StepSettings;
    job?: Job;
    serverId: string;
    server?: Server;
};

export type StepSettings = {
    createBackup?: StepCreateBackupSettings;
};

export type StepCreateBackupSettings = {
    fileName: string;
    backupDefaultExclude: boolean;
    backupIncludeRegex: string;
    backupExcludeRegex: string;
    backupExcludeSystemDatabases: boolean;
    backupIncludeManual: string;
    backupExcludeManual: string;
};

File: src/dotnet/Datack.Common/Models/Data/Step.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Datack.Common.Models.Internal;

namespace Datack.Common.Models.Data
{
    public class Step
    {
        [Key]
        public Guid StepId { get; set; }

        public Guid JobId { get; set; }

        [ForeignKey("JobId")]
        public Job Job { get; set; }

        public String Type { get; set; }

        public String Name { get; set; }

        public String Description { get; set; }

        public Int32 Order { get; set; }

        public StepSettings Settings { get; set; }

        public Guid ServerId { get; set; }

        [ForeignKey("ServerId")]
        public Server Server { get; set; }
    }
}
